static/js/toolbarAlert.js

```diff
@@ -0,0 +1,133 @@
+    var selection = obj.ownerDocument.defaultView.getSelection();
+    var range = obj.ownerDocument.createRange();
+    range.selectNodeContents(obj);
+    selection.removeAllRanges();
+    selection.addRange(range);
+    return this;
+  }
+  
+  jQuery.fn.copyText = function() {
+    var obj = this[0];
+    var ownerDocument = obj.ownerDocument;
+    var textarea = ownerDocument.querySelector('#text_to_copy');
+    textarea.focus({preventScroll: true});
+    textarea.select();
+    ownerDocument.execCommand('copy');
+  }
+  
+  function drawNotifyModal() {
+    if ($padOuter.find('#notifyModal').length === 0) {
+      var baseURL = window.location.href.slice(0, window.location.href.split('/', 3).join('/').length);
+      var onNotify = function(loginOrEmails) {
+        var copiedText = getTextToCopy();
+        $.ajax({
+          method: 'POST',
+          url: baseURL +'/mypads/api/notify-users',
+          data: {
+            auth_token: localStorage.getItem('token'),
+            url: copiedText.url,
+            text: copiedText.text,
+            loginsOrEmails: loginOrEmails
+          },
+          success: function(data, textStatus, jqXHR) {
+            if (data.success === true) {
+              alert('Successfully notified selected users.');
+            } else {
+              alert('Something went wrong.');
+            }
+          },
+          error: function(jqXHR, textStatus, errorThrown) {
+            console.log(errorThrown);
+            alert('Something went wrong.');
+          }
+        });
+      }
+      notifyModal.init(onNotify);
+    }
+    notifyModal.toggle();
+  }
+  
+  function getTextToCopy() {
+    var innerDocWindow = $('iframe[name="ace_outer"]').contents().find('iframe')[0].contentWindow;
+    return {
+      url: window.location.href.slice(0, window.location.href.indexOf('?')) + '?lineNumber=' + (selectedLineNumber + 1),
+      text: innerDocWindow.getSelection().toString()
+    }
+  }
+  
+  function drawToolbarAlert(){
+    var $toolbar = $('.menu_left');
+    $toolbar.append('<li class="separator"></li>');
+    $toolbar.append(
+      '<li id="btnCopyLink" data-type="button">' +
+        '<a class="" title="Copy Link" aria-label="Copy Link">' +
+          '<button class="buttonicon buttonicon-copylink" title="Copy Link" aria-label="Copy Link"></button>' +
+        '</a>' +
+      '</li>'
+    );
+    $toolbar.append(
+      '<li id="btnEmailText" data-type="button">' +
+        '<a class="" title="Email Selected Text" aria-label="Email Selected Text">' +
+          '<button class="buttonicon buttonicon-emailtext" title="Email Selected Text" aria-label="Email Selected Text"></button>' +
+        '</a>' +
+      '</li>'
+    );
+    
+    $toolbar.find('#btnCopyLink').on('click', function(e) {
+      var $textarea = $("<textarea>", {id: "text_to_copy"});
+      var copiedText = getTextToCopy();
+      $textarea.val(copiedText.text + '\n' + copiedText.url);
+      $padOuter.append($textarea);
```

Comment:
Do we need to append the textarea? (I'm not sure about this, but can you try if copytext works even if the textarea is just a floating element not attached to the DOM.